### PR TITLE
Rename actor, store, other refactoring classes

### PR DESCRIPTION
**Pull request by draelos**
Sample actors moved from pytest folder to demos/sample_actors folder. Basic_demo got updated to reflect that but I haven't changed others or confirmed full functionality.

Basic_demo shouldn't be used as a default test case since it requires Caiman, opencv, etc that tend to be annoying to install in the environment.

Nexus and tweak both require a config file, rather than defaulting to basic_demo.yaml.

Store, Actor, Signal got renamed (from Limbo alias to Store, Actor alias to Managed Actor, Spike). ManagedActor uses RunManager and latest 'stop' functionality. Note that all actors (I haven't done a full sweep through demos) need to use the convention of implementing a 'runStep' function which is the function being repeated run in a while loop inside the (managed) 'run' function.  This 'run' also takes care of handling signals so the user doesn't need to. Setup, stop, are by default passed. 

I recommend testers to not just test on the rename_actor branch but also do a local merge (dev <-- rename_actor) and test to confirm the merge goes through alright. With this much refactoring git can get confused. Happy to help with the merge when we get to that point.

### CODE REVIEW

**Comment by jmxpearson** [comment]
@draelos New commit fixes almost all these issues. All `pytest` tests still pass. 

Just please take a look at the one comment renaming `functions` above and resolve, then I can sign off and we can merge.

**Comment by draelos** [comment]
Yep looks good, confirmed tests all passing and things run for me as well!

**Comment by jmxpearson** [comment]
Did you fix the naming of `functions`?

**Comment by draelos** [comment]
Sorry don't know what you mean by naming of functions. Link to the comment?

File: improv/actor.py [hunk 1]
```diff
@@ -161,134 +149,148 @@ def changePriority(self):
...
 
-    @staticmethod
-    def stop_success():
-        return 'stop success'
 
 class RunManager():
     '''
     '''
-    def __init__(self, name, runMethod, setup, q_sig, q_comm, stopMethod, runStore=None):
+    def __init__(self, name, functions, links, runStore=None, timeout=1e-6):
```
[the remark this answers is not in the document]
Comment (jmxpearson) [reply improv/actor.py]: @draelos

**Comment by jmxpearson** [comment]
Line 216 in `actor.py`

**Comment by draelos** [comment]
I feel like I'm missing a comment somewhere, did it get pushed? 
I see that line (https://github.com/project-improv/improv/blob/rename_actor/improv/actor.py#L216) but don't know how you would want the self.functions dict renamed.

**Comment by jmxpearson** [comment]
The name doesn’t tell me anything about what it is/does. Maybe just needs a docstring, but isn’t there more a descriptive name?On Dec 20, 2022, at 19:15, Anne Draelos ***@***.***> wrote:﻿
I feel like I'm missing a comment somewhere, did it get pushed?
I see that line (https://github.com/project-improv/improv/blob/rename_actor/improv/actor.py#L216) but don't know how you would want the self.functions dict renamed.

**Comment by draelos** [comment]
Ah, sure! I had added a brief note in its earlier usage (https://github.com/project-improv/improv/blob/rename_actor/improv/actor.py#L157) but I see how that didn't carry over into the RM class. It's a dict storing the functions of the actor like run, stop, etc, so that the RM can call them given the appropriate signal. Maybe methods is a better name?
Regardless I'll include a more detailed comment/docstring.

**Comment by jmxpearson** [comment]
Unless there's an objection, I'm going to name it `actions`.

**Comment by draelos** [comment]
I quite like actions! I'd say go for it.

**Review by jmxpearson — APPROVED** [review]
Fantastic! Adds a lot to readability.

**Comment by draelos** [comment]
Ah I see these comments got added now.. unsure why not earlier. But it looks good! Thanks for merging.

**Pull request merged**